src/broadcast/broadcast.rs

```diff
@@ -403,9 +406,13 @@ impl<N: NodeIdT> Broadcast<N> {
         let p = p.clone().unwrap();
         let echo_msg = Message::Echo(p.clone());
         let mut step = Step::default();
-        // TODO: iterator not correctly describing right of our_id. Use cycle method on
-        // iterator to correctly specify.
-        let right = self.netinfo.all_ids_except(self.proposer_id()).skip(self.netinfo.num_correct()+self.fault_estimate);
+        // Remaining `2f - g` node ids to the right of our_id (including our_id)
+        // after arranging all node ids in a circular list.
+        let right = self.netinfo.all_ids_except(self.proposer_id())
+                               .cycle()
+                               .skip_while(|x| *x == self.our_id())
+                               .skip(self.netinfo.num_correct() - self.netinfo.num_faulty() + self.fault_estimate)
+                               .take_while(|x| *x == self.our_id());
```

Comment:
This also looks wrong to me. (Also below.)

Reply:
Yes will fix this.